src/webhook/test/webhook.controller.spec.ts

```diff
@@ -22,9 +26,98 @@ describe('WebhookController', () => {
     }).compile();
 
     controller = module.get<WebhookController>(WebhookController);
+    webhookService = module.get<WebhookService>(WebhookService);
+    controller = module.get<WebhookController>(WebhookController);
   });
 
   it('should be defined', () => {
     expect(controller).toBeDefined();
   });
+
+  describe('Create Webhook', () => {
+    const userId = faker.number.int();
+
+    beforeEach(() => {
+      jest
+        .spyOn(webhookService, 'createWebhook')
+        .mockImplementation((_userId: number, _data: CreateWebhookRequest) =>
+          Promise.resolve(createWebhookResponse()),
+        );
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should create a new webhook with token', () => {
+      const data: CreateWebhookRequest = {
+        name: faker.word.noun(),
+        type: WebhookEventType.RecipeCreated,
+        url: faker.internet.url(),
+      };
+
+      controller.createWebhook(userId, data);
+
+      expect(webhookService.createWebhook).toHaveBeenCalled();
+    });
+
+    it('should create a new webhook without token', () => {
+      const data: CreateWebhookRequest = {
+        name: faker.word.noun(),
+        type: WebhookEventType.RecipeDeleted,
+        url: faker.internet.url(),
+        token: faker.string.alphanumeric(32),
+      };
+
+      controller.createWebhook(userId, data);
+
+      expect(webhookService.createWebhook).toHaveBeenCalled();
+    });
+  });
+
+  describe('Delete webhook', () => {
+    it('should delete a token given id', () => {
+      jest
+        .spyOn(WebhookService.prototype, 'deleteWebhook')
+        .mockImplementation((_userId: number, _webhookId: number) =>
+          Promise.resolve(),
+        );
+
+      const userId = faker.number.int();
+      const webhookId = faker.number.int();
+
+      controller.deleteWebhook(userId, webhookId);
+
+      expect(webhookService.deleteWebhook).toHaveBeenCalled();
+
+      jest.clearAllMocks();
```

Comment:
It's safer to move `jest.clearAllMocks();` to `afterEach` method. Imagine if `expect(webhookService.deleteWebhook).toHaveBeenCalled();` fails in this test - `jest.clearAllMocks();` won't be called and other tests might be affected